HW1/P3/P3.py

```diff
@@ -0,0 +1,47 @@
+import findspark
+findspark.init()
+import pyspark
+
+# shut down the previous spark context
+sc.stop() 
+sc = pyspark.SparkContext(appName="myAppName")
+import numpy as np
+import matplotlib.pyplot as plt 
+import matplotlib.cm as cm
+import pdb
+import time
+import pandas as pd
+import itertools
+
+start = time.time()
+
+# Load dictionary
+
+wlist = sc.textFile('EOWL_words.txt',  use_unicode=True)
+
+# convert into lower case words
+wlist = wlist.map(lambda r: r.lower())
+
+# sort all the words alphabetically, check if they are in the dictionary
+anagram = wlist.map(lambda r: ( ''.join( sorted(r) ), r)).groupByKey().sortByKey()
+
+# create the requested structure
+myrdd = anagram.map(lambda r: (r[0], len(r[1]), r[1] ))
+
+# Get the largest valid number of anagrams
+myanagrams = myrdd.takeOrdered(1 ,key=lambda x: -x[1] )
```

Comment:
There are actually multiple anagrams with that maximum number of permutations so you should take more than 1 to make sure you are aware of all maximum anagrams